src/data.rs

```diff
@@ -1259,7 +1259,7 @@ impl<'a> DataNodeRef<'a> {
             panic!("Failed to generate path of the data node");
         }
 
-        char_ptr_to_string(buf.as_ptr())
+        char_ptr_to_string(buf.as_ptr(), true)
```

Comment:
This should be `false` because we're passing a local buffer to `lyd_path()`. This is done for performance reasons.